### ISSUE TITLE
If the user disable "Follow position", the marker(s) are placed in middle of map

### ISSUE THREAD

**Issue by Tripadus** [lmachens/aeternum-map#239]
### **Pre-requisites**
- Have New World & a browser with Aeternum Map opened

### **Steps to reproduce:**
- Have "Follow position" enabled
- Move a certain distance (observe that the map is following and update your position)
- Disable "Follow position"
- Move a certain distance (observer that the map is updating your position - without the map following your position)
- Add a Resource (observe that the marker will be placed in the center of the map [and no, it has nothing to deal with the fact, there it was the character])

_**And he is how I can prove it...**_
- Move your cursor on the map on another position (search for another place, other than you are in right now)
- Add a Resource (observe that the same issue as before happen - the marker will be placed in the middle on the map)

## _Notes_
- Any way you try to reproduce this issue, the outcome will be the same 
- The marker is placed exactly at the middle of the map (depends on how much zoom, resize you have on the browser)
- Here you will have a [video](https://www.youtube.com/watch?v=s5eSS4oIjqo) with timestamps, so you will see better each step

**Comment by lmachens** [comment]
So you expect that new markers are always at the player position?
I think this should only be the case if you follow the player position. If not, you might want to add a marker somewhere else.

**Comment by Tripadus** [comment]
> So you expect that new markers are always at the player position? I think this should only be the case if you follow the player position. If not, you might want to add a marker somewhere else.

Indeed, the marker should be all-time placed on the actual player position.
If you disable by mistake the "Follow Position" or search for something else, in that period of time (on another place, that you currently are, the marker will be placed somewhere else totally)

I find myself in that situation, I work from time to time when I found something that is not on the map (I just add market) when also doing something else...
There are multiple ways this might go wrong.
Another example is, I farm a route and when I focus only on it, I disable "Follow position" and set the map for a certain view.
It's up to you, be you will find yourself, with some markers on the map, without having any relevance into the game (as I do, when I placed it on a point and found out this issue)

**Comment by Sturmfunke** [comment]
How it is right now is "working as intended" I think and it totally makes sense in different scenarios.
Sometimes I want to add a marker at a position where I am not standing right now. It doesn't happen often, but with map updates where things change it is very helpful.
If I start the marker creation process and the marker still gets placed at my last location, it would be really bad usability if I have to move this marker now all the way across the map to the desired location. Also, if I have "follow position" disabled, the marker would be placed at the last known player location, which is most likely too old and therefore irrelevant.

In my opinion, it doesn't make any sense to place a marker at player location if I moved the map away from my location **or** have follow position disabled.

**Comment by Tripadus** [comment]
> How it is right now is "working as intended" I think and it totally makes sense in different scenarios. Sometimes I want to add a marker at a position where I am not standing right now. It doesn't happen often, but with map updates where things change it is very helpful. Imagine I want to place a marker at a location where I am not standing right now and I start the marker creation process and suddenly the map moves away from that location and the marker gets placed at my last known location. It would be really bad usability if I have to move this marker now all the way across the map to the desired location. Also, if I have "follow position" disabled, the marker would be placed at the last known player location, which is most likely too old and therefore most likely not the location I really want to place that marker.
> 
> In my opinion, it doesn't make any sense to place a marker at player location if I moved the map away from my location **or** have follow position disabled.
> 
> Also, if I had follow position disabled by accident and want to place a marker, I will look at the center of the map in the moment I start the marker creation process and then realizing "wait a minute... this area looks weird, that's not where I am". If you dont realize that, I think that is a niche situation. But as I said... even in that situation, placing the marker at the player location makes no sense, as this location is not where I am right now, because follow position is off.
> 
> > Another example is, I farm a route and when I focus only on it, I disable "Follow position" and set the map for a certain view.
> 
> Hm, I never had the situation where I want to have a certain area opened up on the map while I continue playing. Either I stop moving and move around the map, and when I am done, continue playing and the map automatically switches back to my position... or I disable follow position intentionally (which I never felt the need to do in my >700hrs play time) - and enable it again if I want to add a marker at my position, because the map first needs to update my current location anyway before it would make sense to place a marker at the location of the player icon.

I will prefer to have an option, that will let us choose how it will work.
As you see, i see it different :)

**Comment by Sturmfunke** [comment]
Maybe there is a misunderstanding. In what use cases would you want to add
a marker at your last known position before you turned off the live
position?

Tripadus ***@***.***> schrieb am So., 26. Dez. 2021, 14:34:

**Comment by Tripadus** [comment]
> Edit: I think I misunderstood something. I thought you had live position turned off, meaning the app would only know your last position in the moment you turned live postion off. And I coulnt think about a use case where you want to have a marker set at an old player position. But you were only talking about the map not keeping the player position centered. So there are these following scenarios: - S1 Follow position turned ON, character stopped moving and I DIDN'T move the map afterwards -> Marker should be set in the middle of the map, which is the current player location. - S2 Follow position turned ON, character stopped moving and I DID move the map afterwards -> Marker should be set in the middle of the map, which is NOT the current player location. - S3 Follow position turned OFF, but I DIDN'T move the map afterwards, but moved the character meanwhile, stopped, and I DIDN'T move the map after that either -> Map does focus on the player location, marker should be set in the middle of the map, which IS the current player location. (Because I didn't move the map to a certain position.) - S4 Follow position turned OFF, but I DID move the map afterwards, but moved the character meanwhile, stopped, and I DIDN'T move the map after that-> Map does focus on the player location, marker should be set in the middle of the map, which is the current player location (?). (Because if I wanted to place the marker at the location where I moved the map previously before I continued playing, I probably already placed a marker there before moving on ingame. On the other hand, this is not transparent behavior. The user might get confused. Sometimes the marker gets set at current map location, sometimes at current player location. It only takes a little step in the game to result in a different marker creation behavior). This is probably what you @Tripadus meant? - S5 Follow position turned OFF, but moved the character meanwhile, stopped, and I DID move the map after that. -> Map should NOT focus on the player location, but stays at the location I moved the map at, and sets a marker there. (Just like Scenario S2)


I was talking about, the fact, that, the marker should all time have the player position, even with "Follow Position" on/off.
Cause currently, there are like that:
1) "Follow Position" ON = Centered Player = Centered Marker on map (as intended)
2) "Follow Position" OFF = Uncentered Player = Centered Marker on map (as Leon said, as intended, but I personally don't like it)
3) Have you looking somewhere else on the map, then where you are currently with your character, even with "Follow Position" ON/OFF (and not moving your in-game character) = The Marker will be Centered on the map (depends on Zoom, resize)

I also told some things, that might be wrong in my opinion, that people may not know to use well the functionality on the map, and place markers, that have no relevance (with anything) to the actual in-game items

It's not about old player position, it's about the marker is available on the middle of the map (intended) and it's kinda weird for me since I wanna be able to set up and certain area and move in that, without having my map moving, following my position...

Cause if you look the other way, if I have "Follow Position" disabled when I try to put a marker, I need to make one of those 2 ways (considering I follow what I said earlier)
1. Moving myself the marker, where I want it (Zooming In - Accurately place it), then Zooming Out, trying to repair my initial setup
2. Turning on "Follow Position", moving around and coming back to the spot I want to mark, finishing that, then turning off "Follow Position" and trying to repair my initial setup

Instead of letting us decide... (if the marker should be in the middle of the map, or having it all time based on the player position with/without "Follow Position")


> So you expect that new markers are always at the player position? I think this should only be the case if you follow the player position. If not, you might want to add a marker somewhere else.

As in addition, to what I replied earlier to this comment
It will not be stuck on the player position, it will be like other markers, but available at begging from the currently in-game player position, with the new setting turned on
It's more simple than that, you can let us choose how it should work...
It will not be the marker glued to the player position, just an addition, that can be as others, that you implemented
I would like to have "Follow Position" OFF and still walking in a certain area (with zoom out) and not being forced to use "Follow Position", just to have my player position centered, just to have the marker centered also...
As the last mention, add an option, that people can enable it "Added markers only on Player Position", it should not be that hard to do it, since you maintain the markers on the center of the map (as intended, how you specified)
